ocean_lib/ocean/ocean.py

```diff
@@ -164,11 +165,24 @@ def create_nft_token(
             from_wallet=from_wallet,
         )
 
-        address = nft_factory.get_token_address(tx_id)
+        address = nft_factory.get_nft_address(tx_id)
         assert address, "new NFT token has no address"
         token = ERC721Token(self.web3, address)
         return token
 
+    def create_data_token(
```

Comment:
I don't particularly like this placement, it takes a toll on single responsibility principle. I would create a function in erc721 called create_erc20_token, which calls on create_erc20, processes the tx and returns the object instead of the tx. This means in the readme you can call on that function directly, since it has nothing to do with Ocean class itself. Think about it: you are putting code in the Factory to identify things related to ERC721 and ERC20, and you are adding a function to Ocean.

Reply:
Agreed. Thank you!

Reply:
Actually @calina-c , I implemented this and it's not working because the ERC721Factory catches the events, not the ERC721 token. If I want to retrieve the logs, the list will be empty when the ERC721 token calls `get_event_logs`. I think we should keep both in ERC721Factory since we want to retrieve the address by events.